tests/testsuite/build.rs

```diff
@@ -5335,7 +5335,11 @@ fn no_linkable_target() {
     p.cargo("build")
         .with_stderr_data(str![[r#"
 [LOCKING] 1 package to latest compatible version
-[WARNING] The package `the_lib` provides no linkable target. The compiler might raise an error while compiling `foo`. Consider adding 'dylib' or 'rlib' to key `crate-type` in `the_lib`'s Cargo.toml. This warning might turn into a hard error in the future.
+[WARNING] the package `the_lib` provides no linkable target
+  |
+  = [NOTE] this might cause foo to fail compilation
```

Comment:
Should we quote  `foo` with backticks?  Thats something we're inconsistent about